pkg/streamingpromql/operators/functions/sort_by_label.go

```diff
@@ -53,63 +52,83 @@ func (s *SortByLabel) SeriesMetadata(ctx context.Context) ([]types.SeriesMetadat
 		return nil, err
 	}
 
-	// Create a mapping of each series to its original position before sorting.
-	s.originalPositions = make(map[uint64]int, len(innerMetadata))
-	for i, series := range innerMetadata {
-		s.originalPositions[series.Labels.Hash()] = i
+	// Store the original indexes of each piece of metadata and then sort them alongside
+	// the metadata. This allows us to return series in sorted order when requesting them
+	// from the inner operator by their original index.
+	indexes := make([]int, len(innerMetadata))
```

Comment:
We should be able to take this slice from a pool (and we'd then need to return it to the pool in `Close`)